libraries/botbuilder-dialogs-adaptive/src/generators/languageGeneratorManager.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Class which manages cache of all LG resources from a ResourceExplorer. 
+ * This class automatically updates the cache when resource change events occure.
+ */
+import { IResource, ResourceExplorer, FileResource } from 'botbuilder-dialogs-declarative';
+import { MultiLanguageResourceLoader } from '../multiLanguageResourceLoader';
+import { LanguageGenerator } from '../languageGenerator';
+import { TemplateEngineLanguageGenerator } from './templateEngineLanguageGenerator';
+import { normalize, basename} from 'path';
+import { ImportResolverDelegate } from 'botbuilder-lg';
+
+export class LanguageGeneratorManager {
+    private _resourceExporer: ResourceExplorer;
+    
+    /// <summary>
+    /// multi language lg resources. en -> [resourcelist].
+    /// </summary>
+    private _multiLanguageResources: Map<string, IResource[]>;
+
+    public constructor(resourceManager: ResourceExplorer) {
+        this._resourceExporer = resourceManager;
+
+    }
+    // load all LG resources
+    public async loadResources(): Promise<void> {
+        const resources = await this._resourceExporer.getResources('lg');
+        for (const resource of resources) {
+            const generator = await this.getTemplateEngineLanguageGenerator(resource);
+            this._languageGenerator.set(resource.id().toLocaleLowerCase(), generator);
+        }
+    }
+    
+    public _languageGenerator: Map<string, LanguageGenerator> = new Map<string, LanguageGenerator>();
```

Comment:
>public _languageGenerator [](start = 4, length = 25)

why is this public with an '_' prefix?  Underscore should only be used for private members if at all.